### PR TITLE
Take input striding for conv fusion op  based on eager output

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88988
* __->__ #88864

As https://github.com/pytorch/pytorch/pull/88706, we also change the input stride check using eager output.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -3346,59 +3346,47 @@ def _prepare_convolution_fusion_create(
...
-
-    if output_layout_str == "torch.channels_last":
-        stride_order = [0] + list(reversed(range(1, len(kernel_size) + 1)))
-        if len(stride_order) < len(output_size):
-            # add batch dim if it exists
-            stride_order = [len(stride_order)] + stride_order
-    else:
-        stride_order = list(reversed(range(len(output_size))))
+    x = cls.require_stride_order(x, req_stride_order)
+    weight = cls.require_stride_order(weight, req_stride_order)
```
Comment (jgong5) [inline torch/_inductor/ir.py:3373]: Should keep weight layout as in https://github.com/pytorch/pytorch/pull/88891?

Comment (XiaobingSuper) [reply torch/_inductor/ir.py]: Yes, changed.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**